### PR TITLE
feat(pages): Join route handle

### LINKED ISSUE #183: /join/ route handler on web part
## Problem:
Now, when we visit join route specified on note-settings page, it returns 404 status code, and user does not join team of the note
## Solution
Create /join/ route handler on web part

### PR DESCRIPTION
**Pull request by e11sy**
- added team service and repository, also useTeam app service
- now teamMember entity contains id of the note, team is related to
- added join page
* join page shows auth popup if user is not authorized
* join page redirects to note those team user joined if user is authorized
* join page shows linkExpired message if link is expired

resolves #183

### CODE REVIEW

File: src/presentation/pages/Join.vue [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+     * Check if we got id of the note to redirect
+     */
+    if (teamMember.value?.noteId) {
+      router.push(`/note/${teamMember.value?.noteId}`);
+    } else {
+      /**
+       * Join made teamMember null
+       * Means that join link is expired
+       */
+      message.value = t('join.linkExpired');
```
Comment (slaveeks) [inline src/presentation/pages/Join.vue:51]: Maybe redirect to `Home`, for example?

File: src/presentation/pages/Join.vue [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+  } else {
+    /**
+     * Authorize if not authorized yet
+     */
+    showGoogleAuthPopup();
```
Comment (neSpecc) [inline src/presentation/pages/Join.vue:57]: should not be showed when you're visiting own link

File: src/presentation/pages/Join.vue [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  /**
+   * Check if user is authorized
+   */
+  if (user.value !== null) {
+    teamMember.value = await joinNoteTeamByHash(props.invitationHash);
+    /**
+     * Check if we got id of the note to redirect
+     */
+    if (teamMember.value?.noteId) {
+      router.push(`/note/${teamMember.value?.noteId}`);
```
Comment (neSpecc) [inline src/presentation/pages/Join.vue:45]: leave a todo for showing a success alert

File: src/application/services/useTeam.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+export default function useTeam(): useTeamComposableState {
+  /**
+   * Function for adding user to note's team by invitation hash
+   * @param hash - invitation hash of the certain note
+   * @returns Team member or null
+   */
+  async function joinNoteTeamByHash(hash: InvitationHash): Promise<TeamMember | null> {
+    return await teamService.joinNoteTeamByHash(hash);
```
Comment (neSpecc) [inline src/application/services/useTeam.ts:21]: errors are not handled

File: src/presentation/pages/Join.vue [hunk 5]
```diff
@@ -0,0 +1,70 @@
+<template>
+  <Heading :level="1">
```
Comment (neSpecc) [inline src/presentation/pages/Join.vue:2]: align to center and remove heading

File: src/presentation/pages/Join.vue (hunk 1 above)
Comment (e11sy) [reply src/presentation/pages/Join.vue:51]: i think user should see this message and could click to home by himself

**Review by slaveeks — APPROVED** [review]

**Review by neSpecc — APPROVED** [review]

**Pull request merged**